plugins/http_plugin/include/eosio/http_plugin/beast_http_listener.hpp

```diff
@@ -107,7 +109,13 @@ class beast_http_listener : public std::enable_shared_from_this<beast_http_liste
       auto self = this->shared_from_this();
       acceptor_.async_accept(socket_, [self](beast::error_code ec) {
          if(ec) {
-            fail(ec, "accept", self->plugin_state_->logger, "closing connection");
+            if (ec == boost::system::errc::too_many_files_open) {
+               fail(ec, "accept", self->plugin_state_->logger, "too many files open - waiting 500ms");
+               self->timer_.expires_from_now(boost::posix_time::milliseconds(500));
+               self->timer_.wait();
```

Comment:
This needs to be an `async_wait()` -- the synchronous `wait()` is going to block the thread for the duration of the timeout which is a problem especially if running only a single http thread

Reply:
Actually, what we want to do is block the thread that does call `accept()`, so that it doesn't accept any new connections for 500ms. Can you elaborate on why doing a synchronous `wait()` would be an issue?

Reply:
Blocking the thread blocks the thread from performing other work. In the case of there only being a single HTTP thread, that means blocking the HTTP code from performing _any_ HTTP work: writing replies, reading requests, handling closed connections, etc

Reply:
Thanks Matt, I guess I missed the fact that async_accept() was nonblocking. I'll update to an async wait.

Reply:
@spoonincode I replaced the synchronous `wait()` with `async_wait()`